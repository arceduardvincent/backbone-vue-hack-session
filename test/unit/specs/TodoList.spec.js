var { TodoListView } = require('@/views');
var { TodoCollection } = require('@/models');

describe('TodoListView render method', () => {
  it('should render an empty list when empty', () => {
    var todoCollection = new TodoCollection();
    var todoListView = new TodoListView({ collection: todoCollection });
    todoListView.render();
    var html = todoListView.$el[0].outerHTML;
    expect(html).toMatchSnapshot();
  });
  it('should render a list when not empty', () => {
    var todoCollection = new TodoCollection([
      {
        completed: true,
        description: 'A completed task',
      },
      {
        completed: false,
        description: 'An incompete task',
      }
    ]);
    var todoListView = new TodoListView({ collection: todoCollection });
    todoListView.render();
    var html = todoListView.$el[0].outerHTML;
    expect(html).toMatchSnapshot();
  });
});
