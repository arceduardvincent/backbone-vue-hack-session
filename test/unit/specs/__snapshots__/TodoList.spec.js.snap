// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListView render method should render a list when not empty 1`] = `
<div class="todo-list">
    <section id="app" class="todo-container">
        <h2 class="todo-headline">Todos</h2>
        <div class="todo-init">

            <input id="todo-create" class="todo-input" type="text" placeholder="What needs to be done?" autofocus="">

            <span id="todo-toggle-all" class="todo-toggle todo-toggle-all glyphicon glyphicon-ok-sign" title="Toggle all todos"></span>

            <span id="todo-remove-all" class="todo-remove todo-remove-all glyphicon glyphicon-remove-circle" title="Remove all completed todos"></span>
        </div>

        <ul id="todo-list" class="todo-list">
            <li class="todo"> <span class="todo-toggle  todo-toggle-complete  glyphicon glyphicon-ok"></span>
                <input class="todo-input todo-edit" type="text" value="A completed task">
                <label class="todo-desc">A completed task</label>
                <span class="todo-remove glyphicon glyphicon-remove"></span>

            </li>
            <li class="todo"> <span class="todo-toggle  glyphicon glyphicon-ok"></span>
                <input class="todo-input todo-edit" type="text" value="An incompete task">
                <label class="todo-desc">An incompete task</label>
                <span class="todo-remove glyphicon glyphicon-remove"></span>

            </li>
        </ul>
        <footer class="todo-footer">
            <table id="todo-stats" class="todo-stats">
                <tbody>
                    <tr>
                        <th>Total</th>
                        <th>Completed</th>
                        <th>Incomplete</th>
                    </tr>
                    <tr>
                        <td>2</td>
                        <td>1</td>
                        <td>1</td>
                    </tr>
                </tbody>
            </table>
        </footer>
    </section>
</div>
`;

exports[`TodoListView render method should render an empty list when empty 1`] = `
<div class="todo-list">
    <section id="app" class="todo-container">
        <h2 class="todo-headline">Todos</h2>
        <div class="todo-init">

            <input id="todo-create" class="todo-input" type="text" placeholder="What needs to be done?" autofocus="">

            <span id="todo-toggle-all" class="todo-toggle todo-toggle-all glyphicon glyphicon-ok-sign" title="Toggle all todos"></span>

            <span id="todo-remove-all" class="todo-remove todo-remove-all glyphicon glyphicon-remove-circle" title="Remove all completed todos"></span>
        </div>

        <ul id="todo-list" class="todo-list"></ul>
        <footer class="todo-footer">
            <table id="todo-stats" class="todo-stats">
                <tbody>
                    <tr>
                        <th>Total</th>
                        <th>Completed</th>
                        <th>Incomplete</th>
                    </tr>
                    <tr>
                        <td>0</td>
                        <td>0</td>
                        <td>0</td>
                    </tr>
                </tbody>
            </table>
        </footer>
    </section>
</div>
`;
